rusk-abi/tests/lib.rs

```diff
@@ -129,8 +129,78 @@ fn schnorr_signature() {
     );
 }
 
+#[derive(Debug)]
+pub struct TestCircuit {
+    pub a: BlsScalar,
+    pub b: BlsScalar,
+    pub c: BlsScalar,
+}
+
+impl TestCircuit {
+    pub fn new(a: u64, b: u64) -> Self {
+        let a = a.into();
+        let b = b.into();
+        let c = a + b;
+
+        Self { a, b, c }
+    }
+}
+
+impl Circuit for TestCircuit {
+    const CIRCUIT_ID: [u8; 32] = [0xff; 32];
+
+    fn gadget(&mut self, composer: &mut StandardComposer) -> Result<(), Error> {
+        let zero =
+            composer.add_witness_to_circuit_description(BlsScalar::zero());
+        let a = composer.add_input(self.a);
+        let b = composer.add_input(self.b);
+
+        // Make first constraint a + b = c
+        composer.poly_gate(
+            a,
+            b,
+            zero,
+            BlsScalar::zero(),
+            BlsScalar::one(),
+            BlsScalar::one(),
+            BlsScalar::zero(),
+            BlsScalar::zero(),
+            Some(-self.c),
+        );
+
+        Ok(())
+    }
+
+    fn padded_circuit_size(&self) -> usize {
+        1 << 3
+    }
+}
+
 #[test]
 fn verify_proof() {
+    let mut circuit = TestCircuit::new(1, 2);
+
+    let label = TRANSCRIPT_LABEL;
```

Comment:
As mentioned, this is wrong: rusk-abi shouldn't be the "source" of the transcript label, it's actually the opposite. The `rusk-abi` module should be feed with the transcript label (this, or we just hardcoded).

Reply:
But its not clear in that case who is going to feed `rusk-abi` with that